### PR TITLE
Tune layout and painting for split view

### PR DESCRIPTION
**Pull request by sangwoo108**
* Lay out tabs in split view side by side in vertical tabs
* Adjust tile background painting logic.

This change makes drag and drop in vertical tabs awkward. It should be revisited in  follow-ups.

image

Resolves https://github.com/brave/brave-browser/issues/37539

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: chromium_src/chrome/browser/ui/views/tabs/tab_strip_layout_helper.cc [hunk 1]
```diff
@@ -4,13 +4,15 @@
...
 #include "brave/browser/ui/views/tabs/brave_tab_strip.h"
 
-#define CalculateTabBounds                                                     \
-  use_vertical_tabs_&& FillGroupInfo(tab_widths)                               \
-      ? tabs::CalculateVerticalTabBounds(layout_constants, tab_widths,         \
-                                         tabstrip_width,                       \
-                                         tab_strip_->IsVerticalTabsFloating()) \
+#define CalculateTabBounds                                \
+  FillTiledState(tab_widths, tab_strip_) &&               \
+          use_vertical_tabs_&& FillGroupInfo(tab_widths)  \
```
Comment (simonhong) [inline chromium_src/chrome/browser/ui/views/tabs/tab_strip_layout_helper.cc:12]: nit:
```suggestion
          use_vertical_tabs_ && FillGroupInfo(tab_widths)  \
```

**Comment by simonhong** [comment]
In vertical tab mode, layout is broken when there is a group.
Below is - collapsed group is top most. and below tiles layout is broken.
image

**Comment by sangwoo108** [comment]
> In vertical tab mode, layout is broken when there is a group. Below is - collapsed group is top most. and below tiles layout is broken. image

Yes, Tab group is not considered yet.

**Review by simonhong — APPROVED** [review]
++ 👍🏼

File: chromium_src/chrome/browser/ui/views/tabs/tab_strip_layout_helper.cc (hunk 1 above)
Comment (sangwoo108) [reply chromium_src/chrome/browser/ui/views/tabs/tab_strip_layout_helper.cc:12]: The formatter keeps removing the whitespace 🥲

Comment (simonhong) [reply chromium_src/chrome/browser/ui/views/tabs/tab_strip_layout_helper.cc:12]: Oh, i see.

File: chromium_src/chrome/browser/ui/views/tabs/tab_slot_controller.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_CHROME_BROWSER_UI_VIEWS_TABS_TAB_SLOT_CONTROLLER_H_
+#define BRAVE_CHROMIUM_SRC_CHROME_BROWSER_UI_VIEWS_TABS_TAB_SLOT_CONTROLLER_H_
+
+#define IsGroupCollapsed                               \
+  IsTabTiled(const Tab* tab) const;                    \
+  virtual bool IsFirstTabInTile(const Tab* tab) const; \
```
[the remark this answers is not in the document]
Comment (sangwoo108) [reply chromium_src/chrome/browser/ui/views/tabs/tab_slot_controller.h]: Done. Added FakeTabStripController override together.

**Review by mkarolin — APPROVED** [review]
`chromium_src` LGTM

**Comment by sangwoo108** [comment]
The latest design changed. This will be partially reverted.

**Pull request merged**